### ISSUE TITLE
[java] CloseResource: false positive with lambda expression

### ISSUE THREAD

**Issue by Han-Cui** [pmd/pmd#4624]
**Affects PMD Version:** 7.0.0

**Rule:** CloseResource

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#closeresource>

**Description:**
When _c.close()_ is in a lambda expression, PMD ignores it, and warns in line 7.
**Code Sample demonstrating the issue:**
Maybe a false positive:

```java
import java.sql.Connection;
import net.sourceforge.pmd.lang.java.rule.errorprone.closeresource.Pool;

public class Foo {

    void bar(Pool pool) {
        Connection c = pool.getConnection();
        try {
            Runnable closeConnection = () -> {
                try {
                    c.close();
                } catch (Exception ex) {
                    // ...
                }
            };

            // ...

            closeConnection.run();
        } catch (Exception e) {
            // ...
        }
    }

}
```

**Expected outcome:**
No warnings.

PMD reports a violation at line 7, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
Hi @Han-Cui. As your code stands, this is not a false positive. An exception may occur in the `try` before `closeConnection.run()` and the connection would not be closed.

Moving the close to a `finally` block however would be a false positive identical to #4623 (it's the same scenario)

I'm closing this and updating #4623. Feel free to comment if you have reason to believe this is still a FP as it stands.

**Comment by Han-Cui** [comment]
I agree, thank for your kind and prompt reply!
